CorsixTH/Src/th_movie.cpp

```diff
@@ -166,24 +145,26 @@ void THMoviePictureBuffer::allocate(SDL_Renderer *pRenderer, int iWidth, int iHe
     //Do not change m_iWriteIndex, it's used by the other thread.
     //m_iReadIndex is only used in this thread.
     m_iReadIndex = m_iWriteIndex;
-    SDL_LockMutex(m_pMutex);
+
+    std::lock_guard<std::mutex> lock(m_mutex);
     m_iCount = 0;
     m_fAllocated = true;
-    SDL_CondSignal(m_pCond);
-    SDL_UnlockMutex(m_pMutex);
+    m_cond.notify_one();
 }
 
 void THMoviePictureBuffer::deallocate()
 {
-    SDL_LockMutex(m_pMutex);
-    m_fAllocated = false;
-    SDL_UnlockMutex(m_pMutex);
+    {
+        std::lock_guard<std::mutex> lock(m_mutex);
+        m_fAllocated = false;
+    }
+
     for(int i = 0; i < ms_pictureBufferSize; i++)
     {
-        SDL_LockMutex(m_aPictureQueue[i].m_pMutex);
+        std::lock_guard<std::mutex> pictureLock(m_aPictureQueue->m_mutex);
```

Comment:
Weird that you don't need a [i] here, while the previous version had one, and line 256 also seems to have one